src/cli.rs

```diff
@@ -368,6 +368,33 @@ impl PyPiAddOpts {
     }
 }
 
+#[derive(Debug, Parser)]
+pub struct TarballAddOpts {
+    /// Lockable URL
+    pub url: Url,
+}
+
+impl TarballAddOpts {
+    pub fn add(&self) -> Result<(String, Pin)> {
+        // TODO this assumes common forgejo/github/... URL scheme of:
```

Comment:
this is a bit i'm really not sure about, what do you think?

Reply:
Yeah I already stumbled upon this in the earlier draft, forgot to comment on it. I'm a bit torn, because it will kinda work as a good default most of the time, but when it won't it will be confusing and generally a bad experience. Therefore I'm inclined to remove that part and always return `None` for the name instead

Reply:
ok, i can definitely do that. is it ok if i change all the other function impls to return `Option<String>` instead of `String` too to make it consistent, or should i rather work around this one being out of line [here](https://github.com/andir/npins/pull/119/files#diff-b2812f19576dd53d0c35b107a322f58a00fce6977f4b5976e1961853982af3ccR441-R448)?

Reply:
Technically, the downstream caller checks for an empty string, so simply passing `"".into()` would technically work. Doing a small refactoring would be cleaner, but I can do it as a follow-up instead if you prefer. It's not urgent

Reply:
changed it to always ask and did the small refactor for the rest of them^^